src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -2067,4 +1903,83 @@ private void updateGroupSource(final List<Integer> gids, final Integer newGroupS
 		}
 	}
 
+
+	public List<Variable> getGermplasmAttributeVariables(final List<Integer> gids, final String programUUID) {
+		try {
+			final SQLQuery sqlQuery = this.getSession().createSQLQuery("select "
+				+ " v.cvterm_id AS " + GermplasmDAO.VARIABLE_ID + ", "  //
+				+ " v.name AS " + GermplasmDAO.VARIABLE_NAME + ", "  //
+				+ " v.definition AS " + GermplasmDAO.VARIABLE_DEFINITION + ", "  //
+				+ " vmr.mid AS "  + GermplasmDAO.METHOD_ID + ", "  //
+				+ " vmr.mn AS "  + GermplasmDAO.METHOD_NAME + ", "  //
+				+ " vmr.md AS "  + GermplasmDAO.METHOD_DEFINITION + ", "  //
+				+ " vpr.pid AS "  + GermplasmDAO.PROPERTY_ID + ", "  //
+				+ " vpr.pn AS "  + GermplasmDAO.PROPERTY_NAME + ", "  //
+				+ " vpr.pd AS "  + GermplasmDAO.PROPERTY_DEFINITION + ", "  //
+				+ " vsr.sid AS "  + GermplasmDAO.SCALE_ID + ", "  //
+				+ " vsr.sn AS "  + GermplasmDAO.SCALE_NAME + ", "  //
+				+ " vsr.sd AS " + GermplasmDAO.SCALE_DEFINITION + ", "  //
+				+ " vpo.alias  AS " + GermplasmDAO.VARIABLE_ALIAS + ", "  //
+				+ " vpo.expected_min AS " + GermplasmDAO.VARIABLE_EXPECTED_MIN + ", "  //
+				+ " vpo.expected_max AS " + GermplasmDAO.VARIABLE_EXPECTED_MAX
+				+ " FROM cvterm v INNER JOIN cvtermprop cp ON cp.type_id = " + TermId.VARIABLE_TYPE.getId()
+				+ " and v.cvterm_id = cp.cvterm_id " //
```

Comment:
It's more performant to do the filtering of variable types to passport and attribute as part of main clause instead of where:`
INNER JOIN cvterm varType on varType.name = cp.value AND varType.cvterm_id in (...)`

Reply:
@cuenyad please adreess all GermplasmDAO comments

Reply:
done.